openml/flows/flow.py

```diff
@@ -1,80 +1,352 @@
+            arguments[key] = dic.get("oml:" + key)
+
+        # has to be converted to an int if present and cannot parsed in the
+        # two loops above
+        arguments['flow_id'] = (int(dic['oml:id']) if dic.get("oml:id")
+                                is not None else None)
+
+        # Now parse parts of a flow which can occur multiple times like
+        # parameters, components (subflows) and tags. These can't be tackled
+        # in the loops above because xmltodict returns a dict if such an
+        # entity occurs once, and a list if it occurs multiple times.
+        # Furthermore, they must be treated differently, for example
+        # for components this method is called recursively and
+        # for parameters the actual information is split into two dictionaries
+        # for easier access in python.
+
+        parameters = OrderedDict()
+        parameters_meta_info = OrderedDict()
+        if 'oml:parameter' in dic:
+            # In case of a single parameter, xmltodict returns a dictionary,
+            # otherwise a list.
+            if isinstance(dic['oml:parameter'], dict):
+                oml_parameters = [dic['oml:parameter']]
+            else:
+                oml_parameters = dic['oml:parameter']
+
+            for oml_parameter in oml_parameters:
+                parameter_name = oml_parameter['oml:name']
+                default_value = oml_parameter['oml:default_value']
+                parameters[parameter_name] = default_value
+
+                meta_info = OrderedDict()
+                meta_info['description'] = oml_parameter.get('oml:description')
+                meta_info['data_type'] = oml_parameter.get('oml:data_type')
+                parameters_meta_info[parameter_name] = meta_info
+        arguments['parameters'] = parameters
+        arguments['parameters_meta_info'] = parameters_meta_info
+
+        components = OrderedDict()
+        if 'oml:component' in dic:
+            # In case of a single component xmltodict returns a dict,
+            # otherwise a list.
+            if isinstance(dic['oml:component'], dict):
+                oml_components = [dic['oml:component']]
+            else:
+                oml_components = dic['oml:component']
+
+            for component in oml_components:
+                flow = OpenMLFlow._from_dict(component)
+                components[component['oml:identifier']] = flow
+        arguments['components'] = components
+
+        tags = []
+        if 'oml:tag' in dic and dic['oml:tag'] is not None:
+            # In case of a single tag xmltodict returns a dict, otherwise a list
+            if isinstance(dic['oml:tag'], dict):
+                oml_tags = [dic['oml:tag']]
+            else:
+                oml_tags = dic['oml:tag']
+
+            for tag in oml_tags:
+                tags.append(tag)
+        arguments['tags'] = tags
+
+        arguments['model'] = None
+        return cls(**arguments)
+
+    def __eq__(self, other):
```

Comment:
I'm still wondering where we need that. Can you point that out again? I would rather not do this.